File: Movie-Store/src/app/customer-list/customer.ts
export class Customer {
    customerId! : number;
    storeId!: {
      storeId: number;
      managerStaffId: number;
      addressId: number;
      lastUpdate: string;
    };
    firstName!: string;
    lastName!: string;
    email!: string;
    addressId!: {
      addressId: number;
      cityId: {
          cityId: number;
          countryId: {
          countryId: number;
          country: string;
          lastUpdate: string;
        };
        city: string;
        lastUpdate: string;
      };
      address: string;
      address2: string | null;
      district: string;
      postal_code: string;
      phone: string;
      location: any;
      lastUpdate: string;
    };
    active!: string;
    createDate!: string;
    lastUpdate!: string;
  }
  

File: filmerentalstore/src/main/java/com/sprint/filmerentalstore/entity/Customer.java
package com.sprint.filmerentalstore.entity;

import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalDateTime;

import com.fasterxml.jackson.annotation.JsonFormat;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;



@Entity
@Data
@Table(name="customer")
public class Customer implements Serializable{

    /**
	 * 
	 */
	private static final long serialVersionUID = 1L;

	@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "customer_id")
    private Short customerId;

    @ManyToOne
    @JoinColumn(name = "store_id", referencedColumnName = "store_id")
    private Store storeId;
    
    @ManyToOne
    @JoinColumn(name = "address_id" , referencedColumnName = "address_id")
    private Address addressId;
    
//    @OneToMany(mappedBy="customerId")
//    private List<Rental> rental;

    @Column(name = "firstName")
    private String firstName;

    @Column(name = "last_name")
    private String lastName;

    @Column(name = "email")
    private String email;


    @Column(name = "active")
    private String active;

    @JsonFormat(pattern = "yyyy-MM-dd")
    @Column(name = "create_date")
    private LocalDate createDate;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss.SSSSSS")
    @Column(name = "last_update")
    private LocalDateTime lastUpdate;

    
}
